optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -29,29 +27,33 @@
 import java.lang.reflect.Type;
 import java.math.BigDecimal;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.function.IntSupplier;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import javax.script.ScriptEngine;
-import javax.script.ScriptEngineManager;
-import javax.script.ScriptException;
-
 import org.optaplanner.core.api.domain.lookup.PlanningId;
 import org.optaplanner.core.config.AbstractConfig;
 import org.optaplanner.core.impl.domain.common.AlphabeticMemberComparator;
 import org.optaplanner.core.impl.domain.common.ReflectionHelper;
 import org.optaplanner.core.impl.domain.common.accessor.MemberAccessor;
 import org.optaplanner.core.impl.domain.common.accessor.MemberAccessorFactory;
+import org.slf4j.Logger;
+
+import static org.optaplanner.core.impl.domain.common.accessor.MemberAccessorFactory.MemberAccessorType.FIELD_OR_READ_METHOD;
 
 public class ConfigUtils {
 
+    /**
+     * Name of the variable that represents {@link Runtime#availableProcessors()}.
+     */
+    public static final String THREAD_COUNT_AUTO = "AUTO";
+    public static final String AVAILABLE_PROCESSOR_COUNT = "availableProcessorCount";
```

Comment:
Let's also remove availableProcessorCount:
While "availableProcessorCount / 2" can be usefull (despite no one using it), "availableProcessorCount" is always a bad idea (freezes the computer). It's a loaded gun tied to your leg aimed at your foot. There's no way to miss.